tests/conftest.py

```diff
@@ -418,4 +419,20 @@ def pytest_sessionfinish(session: pytest.Session, exitstatus: int):
     with open("tests/output/proxy_calls.json", encoding="utf-8") as file:
         current = json.load(file)
         if current != filtered_calls:
-            raise AssertionError("API calls have changed, run scripts/snapshot-update")
+            diff = ""
+            for test in current:
+                if test not in filtered_calls:
+                    diff += f"Test '{test}' was removed\n"
+            for test in filtered_calls:
+                if test not in current:
+                    diff += f"Test '{test}' was added\n"
+            for test in filtered_calls:
+                if test not in current:
+                    continue
+                if filtered_calls[test] == current[test]:
+                    continue
+                diff += f"Test '{test}' has changed\n"
+                diff += "\n".join(_compare_eq_iterable(filtered_calls[test], current[test], 3))
+                diff += "\n"
+
+            raise AssertionError(f"API calls have changed, run scripts/snapshot-update\n{diff}")
```

Comment:
This whole function is way to complicated for my taste. Why do we need to check if we need to update the snapshot? Normally the developer that changes a test will update the snapshot if needed.

Reply:
The rationale for the change is to show the diff in CI in case CI behaves different than local.

Reply:
Right, but why do we have this function in the first place?

Reply:
To be able to track increase /decrease in API usage.

Reply:
Why don't we have a specific test that counts this and is added for each feature in that case? This hook runs on every test and has complicated logic as it wants to be generic instead of specific.

Reply:
It runs in the finish ov everything not after each test.
It also provide the host and path that was called and how many times instead of just a simple couny

Reply:
> It runs in the finish ov everything not after each test.

That's what I meant.

If you really need to count these things I suggest trying to use a `Counter` instance to do that instead. Eg by making tuples of the url and path in a list for all calls and then creating the Counter from that.

Side note: I looked at the RequestMocker and how the fixtures are loaded and it looks inefficient since it loads files on the fly inside the event loop instead of separately once for each fixture.